composites/Plugin/SnippetPreview/components/SnippetPreview.js

```diff
@@ -197,41 +199,52 @@ const Amp = styled.div`
  * Highlights a keyword with strong React elements.
  *
  * @param {string} locale ISO 639 (2/3 characters) locale.
- * @param {string} keyword The keyword.
+ * @param {string[]} keywordForms The array of keyword forms.
  * @param {string} text The text in which to highlight a keyword.
  * @param {string} cleanText Optional. The text in which to highlight a keyword
  *                           without special characters and diacritics.
  *
  * @returns {ReactElement} React elements to be rendered.
  */
-function highlightKeyword( locale, keyword, text, cleanText ) {
-	if ( ! keyword ) {
+function highlightKeyword( locale, keywordForms, text, cleanText ) {
+	if ( keywordForms.length === 0 ) {
 		return text;
 	}
 
-	/*
-	 * When a text has been cleaned up from special characters and diacritics
-	 * we need to match against a cleaned up keyword as well.
-	 */
+	keywordForms = uniq( flatten( keywordForms ) );
```

Comment:
This is overwriting the variable passed to this function. I would say as a general rule, do not do this as it could have unexpected consequences.
Thinking more on this, this is the only place these are used right? Maybe it would be better to have the `keywordForms` in the store already be in this format.